src/ansible_compat/config.py

```diff
@@ -0,0 +1,98 @@
+
+from ansible_compat.constants import ANSIBLE_MISSING_RC
+
+options = Namespace(
+    cache_dir=None,
+    colored=True,
+    configured=False,
+    cwd=".",
+    verbosity=False,
+    offline=False,
+    project_dir=".",  # default should be valid folder (do not use None here)
+    extra_vars=None,
+)
+
+
+# Used to store collection list paths (with mock paths if needed)
+collection_list: List[str] = []
+
+
+def get_rule_config(rule_id: str) -> Dict[str, Any]:
+    """Get configurations for the rule ``rule_id``."""
+    rule_config = options.rules.get(rule_id, dict())
+    if not isinstance(rule_config, dict):
+        raise RuntimeError("Invalid rule config for %s: %s" % (rule_id, rule_config))
+    return rule_config
+
+
+@lru_cache()
+def ansible_collections_path() -> str:
+    """Return collection path variable for current version of Ansible."""
+    # respect Ansible behavior, which is to load old name if present
+    for env_var in ["ANSIBLE_COLLECTIONS_PATHS", "ANSIBLE_COLLECTIONS_PATH"]:
+        if env_var in os.environ:
+            return env_var
+
+    # https://github.com/ansible/ansible/pull/70007
+    if ansible_version() >= ansible_version("2.10.0.dev0"):
+        return "ANSIBLE_COLLECTIONS_PATH"
+    return "ANSIBLE_COLLECTIONS_PATHS"
+
+
+def parse_ansible_version(stdout: str) -> Tuple[str, Optional[str]]:
+    """Parse output of 'ansible --version'."""
+    # Ansible can produce extra output before displaying version in debug mode.
+
+    # ansible-core 2.11+: 'ansible [core 2.11.3]'
+    match = re.search(r"^ansible \[(?:core|base) ([^\]]+)\]", stdout, re.MULTILINE)
+    if match:
+        return match.group(1), None
+    # ansible-base 2.10 and Ansible 2.9: 'ansible 2.x.y'
+    match = re.search(r"^ansible ([^\s]+)", stdout, re.MULTILINE)
+    if match:
+        return match.group(1), None
+    return "", "FATAL: Unable parse ansible cli version: %s" % stdout
+
+
+@lru_cache()
+def ansible_version(version: str = "") -> Version:
+    """Return current Version object for Ansible.
+
+    If version is not mentioned, it returns current version as detected.
+    When version argument is mentioned, it return converts the version string
+    to Version object in order to make it usable in comparisons.
+    """
+    if not version:
+        proc = subprocess.run(
+            ["ansible", "--version"],
+            universal_newlines=True,
+            check=False,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        )
+        if proc.returncode == 0:
+            version, error = parse_ansible_version(proc.stdout)
+            if error is not None:
+                print(error)
+                sys.exit(ANSIBLE_MISSING_RC)
+        else:
+            print(
+                "Unable to find a working copy of ansible executable.",
+                proc,
+            )
+            sys.exit(ANSIBLE_MISSING_RC)
+    return Version(version)
+
+
+if ansible_collections_path() in os.environ:
```

Comment:
It's best not to have any import-time code execution. You could use `__getattr__()` to make it on-access though.

Reply:
If you are talking about https://www.python.org/dev/peps/pep-0562/  that is py37+, so not usable for us.

I am aware of this issue and I plan to refector the library to use a manager that can return  property. That would be very useful for other bits inside too. Still, for the moment I will keep the original implementation in this PR.

Reply:
There's a backport: https://pypi.org/project/pep562/